src/jit/flowgraph.cpp

```diff
@@ -8890,6 +8890,138 @@ class MergedReturns
 };
 }
 
+//-------------------------------------------------------------------------
+//  fgExpandBoolReturns: find all Bool BBJ_RETURN basic blocks and expand
+//  them into BBJ_COND if it's possible to re-use existing `return true/false`
+//  blocks. E.g.:
```

Comment:
I believe that it is illegal to jump into a try protected region from outside.
So you need to call `sameTryRegion` before perform this transformation